content/posts/numpy/numpy2/index.md

```diff
@@ -0,0 +1,119 @@
+changes happened over the years, but because of backward compatibility concerns
+they remained largely invisible to the users. NumPy 2.0 is, in part, the
+culmination of these improvements, allowing us to discard some of the legacy
+behaviors or details of the ABI that prevent future improvements.
+
+We started concrete plans for the 2.0 release more than a year ago, at a four hour
+long [public planning meeting](https://github.com/numpy/archive/tree/main/2.0_developer_meeting)
+in April 2023. Many of the key changes were proposed and discussed. The key goals
+we decided on there were perhaps even larger and more ambitious in scope than
+some of us expected. This also unlocked some extra energy - which has been great to see.
+After the meeting and over the course of the last year, NumPy enhancement
+proposals ([NEPs](https://numpy.org/neps/)) for each major change were written,
+reviewed, and implemented.
+
+Some of the key highlights and major changes are:
+
+- Cleaned-up and streamlined Python API ([NEP 52](https://numpy.org/neps/nep-0052-python-api-cleanup.html)):
+  The Python API has undergone a thorough cleanup, making it easier to learn
+  and use NumPy. The main namespace has been reduced by approximately 10%, and
+  the more niche `numpy.lib` namespace has been reduced by about 80%, providing
+  a clearer distinction between public and private API elements.
+
+- Improved scalar promotion rules: The scalar promotion rules have been
+  updated, as proposed in [NEP 50](https://numpy.org/neps/nep-0050-scalar-promotion.html)
+  addressing surprising behaviors in type promotion e.g. with zero dimensional arrays.
+
+- Powerful new DType API and a new string dtype: NumPy 2.0 introduces a new API
+  for implementing user-defined custom data types as proposed by
+  [NEP 41](https://numpy.org/neps/nep-0041-improved-dtype-support.html). We used
+  this new API to implement `StringDType`, offering efficient and painless
+  support for variable length strings which was proposed in
+  [NEP 55](https://numpy.org/neps/nep-0055-string_dtype.html). And it is our hope
+  that enable future new data types with interesting new capabilities in the
+  PyData ecosystem and in NumPy itself.
+
+- Windows compatibility enhancements: The default 32-bit integer representation
+  on Windows has been updated to 64-bit on 64-bit architectures, addressing one
+  of the most common issues a developer runs into trying to get NumPy code to
+  work portably on Windows and Unix-like operating systems.
+
+- Full support for the Python array API standard: This is the first release to
+  include full support for the array API standard (v2022.12), which was enabled
+  by the promotion rules and API cleanup mentioned above, as well as by
+  adding new APIs and aligning existing APIs and behavior with the standard,
+  as proposed by [NEP 56](https://numpy.org/neps/nep-0056-array-api-main-namespace.html).
```

Comment:
```suggestion
- Support for the Python array API standard: This is the first release to
  include full support for the array API standard (v2022.12), made possible
  by the new promotion rules, APIs, and API cleanup mentioned above.
  We also aligned existing APIs and behavior with the standard,
  as proposed in [NEP 56](https://numpy.org/neps/nep-0056-array-api-main-namespace.html).
```

I tried to keep the meaning the same here, but to streamline the wording, since there were a lot of joined sub-sentences.